src/com/tigersale/userInterface/customerUser/ListOfProductView.java

```diff
@@ -0,0 +1,212 @@
+package com.tigersale.userInterface.customerUser;
+
+import com.tigersale.database.ProductTable;
+import com.tigersale.database.ShoppingListTable;
+import com.tigersale.model.CustomerUser;
+import com.tigersale.model.Order;
+import com.tigersale.model.Product;
+import com.tigersale.userInterface.AbstractView;
+import javafx.util.Pair;
+
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * List of Products View
+ * @author My Tran
+ */
+public class ListOfProductView extends AbstractView {
+
+    /**
+     * Constructor for the ListOfProductView
+     * @param scanner The scanner to retrieve user input from
+     */
+    public ListOfProductView(Scanner scanner) {
+        super(scanner);
+    }
+
+    /**
+     * This method acts as the view for the CustomerUser to view the Product list
+     * @param user the working CustomerUser
+     */
+    public void runListOfProductView(CustomerUser user) {
+        while (true) {
+            System.out.println("=========================================================");
+            System.out.println("                     List of Products                    ");
+            System.out.println("=========================================================");
+
+            //Retrieve the list of products from the database
+            List<Product> products = ProductTable.viewProducts();
+            int size = products.size();
+            int choice;
+
+            if (size == 0)
+            {
+                System.out.println("There is no products in the system yet");
+                return;
+            } else
+            {
+                //Print all the products with associated numbers
+                System.out.println("Please choose the number associated with the product to view detail:");
```

Comment:
This needs to give the user the option to go back.  which should be the '0' option